go/enclave/enclave.go

```diff
@@ -109,6 +109,9 @@ func NewEnclave(
 	mgmtContractLib mgmtcontractlib.MgmtContractLib,
 	logger gethlog.Logger,
 ) common.Enclave {
+	jsonConfig, _ := json.MarshalIndent(config, "", "  ")
```

Comment:
I moved this config dump higher up because I was hitting crit errors in the constructor method below before it reached the config log msg, making it hard to know what was going on.

Reply:
there is this library i added, sdumps or whatever that should be better

Reply:
Nice yeah I've seen that around, the `litter` one. Not sure how well it will work with log messages though because of the line breaks, looks like all the current usages are for stdout prints.

But yeah agree we should try it when we get some time cos it is a PITA to read. I'll leave it for this PR though cos was just moving it earlier.